(#25) Move Plotting functionality from TSoG_X1_Sim to TemplateTestCase

Nufflee opened the pull request:
This allows every tests case to have a fully custom plotting function by assigning it to `TestCase.Plot` or a standard one that plots all simulation results by assigning `StandardPlots` to `TestCase.Plot`.

To run the whole simulation now, you have to:
```matlab
run("Testing/startup.m") % Only has to be run once per Octave session

run("Testing/TemplateTestCase.m")
```

Closes #25.

Nufflee committed:
Allow every TestCase to have a custom plotting function and create StandardPlots.

Allows every TestCase to have a fully custom plotting function and adds `StandardPlots` which contains the default plotting function that just plots all simulation resutls.

TheGreatFez commented on Simulation/StandardPlots.m:
```diff
@@ -0,0 +1,27 @@
+  set(groot, 'DefaultLineLineWidth', 5);
+
+  plot(Results.Time, Results.X);
+  hold on;
```
Two questions:
1. Do you need to specify hold on every time? Usually in MATLAB you just need one and if you need to turn it off you have to specify `hold off`
2. If the answer is "no" to 1., do you also need to add `hold off`? This usually prevents weird behavior of overlapping plotting on new calls.

TheGreatFez commented on Simulation/TSoG_X1_Sim.m:
```diff
@@ -27,6 +26,8 @@
                               15 1];
     TestCase.StopTime = 15;
+
+    TestCase.Plot = @StandardPlots;
```
The issue was to remove the plotting from the simulation file entirely. Reason being is if I wanted to create an analysis where I run a large number of tests (or even a couple) then it will be halted or be plotting over eachother every time which is not really useful. 

This line will need to be removed from the simulation file.

TheGreatFez commented on Simulation/TSoG_X1_Sim.m:
```diff
@@ -100,26 +101,5 @@
-
-  grid on;
-endfunction
+  TestCase.Plot(Results)
```
See above comment, this line should also be removed.

TheGreatFez reviewed (changes requested):
I had 2 questions on the Standard Plots and then a couple changes that need to be made on TSoG_X1_Sim

Nufflee committed:
Don't run plotting functions in TSoG_X1_Sim

This is done because we want to be able to run multiple sims in succession without plots interrupting the execution.

* Add Testing/startup.m which sets up the path so tha test cases can find sim functions

Nufflee committed:
Use a single hold on and hold off in StandardPlots.

Replace multiple repeating `hold on` statements with a single `hold on` and `hold off` statement.

Nufflee committed:
Run startup.m in TemplateTestCase.m

This is done so the user doesn't have to separately run it every time they start up Octave.

TheGreatFez reviewed (approved):
Great work!
